src/ripple/basics/Buffer.h

```diff
@@ -34,14 +34,49 @@ namespace ripple {
 class Buffer
 {
 private:
-    std::unique_ptr<
-        std::uint8_t[]> p_;
+    std::unique_ptr<std::uint8_t[]> p_;
     std::size_t size_ = 0;
 
 public:
     Buffer() = default;
-    Buffer (Buffer const&) = delete;
-    Buffer& operator= (Buffer const&) = delete;
+
+    /** Create an uninitialized buffer with the given size. */
+    explicit
+    Buffer (std::size_t size)
+        : p_ (size ? new std::uint8_t[size] : nullptr)
+        , size_ (size)
+    {
+    }
+
+    /** Create a buffer as a copy of existing memory.
+
+        @param data a pointer to the existing memory. If
+                    size is non-zero, it must not be null.
+        @param size size of the existing memory block.
+    */
+    Buffer (void const* data, std::size_t size)
+        : Buffer (size)
+    {
+        if (size)
+            std::memcpy(p_.get(), data, size);
```

Comment:
@ximinez According to: http://en.cppreference.com/w/cpp/string/byte/memcpy
`If either dest or src is a null pointer, the behavior is undefined, even if count is zero.` So I think the check is still needed since data could be  a null pointer in that case.